src/ContentValidationListener.php

```diff
@@ -213,7 +215,35 @@ public function onRoute(MvcEvent $e)
         }
 
         $inputFilter->setData($data);
+
         if ($inputFilter->isValid()) {
+            if (isset($this->config[$controllerService]['use_raw_data'])
+                    && $this->config[$controllerService]['use_raw_data'] === true) {
```

Comment:
One other note: if the key is NOT present, we also have to set the raw data in order to preserve BC for existing services. I'll make that change during merge as well, and add a test for it.